### PR TITLE
Add Java GitHub API

### PR DESCRIPTION
**Pull request by Richie78321**
Add the Java GitHub API and a helper class that loads authorization from environment variables.

### CODE REVIEW

**Comment by billyeh** [comment]
> A Googler has manually verified that the CLAs look good.
> 
> (Googler, please make sure the reason for overriding the CLA status is clearly documented in these comments.)
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogleinterns%2Fstep22-2020%2Fpull%2F30) for more info**.

Github Actions linter (formatting changes only) b/160174261.

File: src/main/java/com/google/opensesame/github/GitHubGetter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    gitHub = new GitHubBuilder().withPassword(clientID, clientSecret).build();
+
+    try {
+      gitHub.checkApiUrlValidity();
+    } catch (IOException e) {
+      System.err.println(
+          "Invalid GitHub credentials. Please refer to the project README to"
+              + " configure this. Using unauthorized GitHub API for now.");
+      gitHub = GitHub.connectAnonymously();
```
Comment (josephdhurst) [inline src/main/java/com/google/opensesame/github/GitHubGetter.java:51]: Setting `gitHub` in getGitHub makes it clearer that `buildGitHubInterface` is only meant to be called once and that the result is cached in gitHub for future use.

```suggestion
  public static GitHub getGitHub() throws IOException {
    if (gitHub == null) {
      gitHub = buildGitHubInterface();
    }

    return gitHub;
  }

  /**
   * Builds the GitHub interface from environment variables.
   *
   * <p>With AppEngine, the environment variables are set in the appengine-web.xml file.
   *
   * @throws IOException Throws IOException if there is an error building the GitHub interface.
   */
  private static GitHub buildGitHubInterface() throws IOException {
    String clientID = System.getenv("GITHUB_CLIENT_ID");
    String clientSecret = System.getenv("GITHUB_CLIENT_SECRET");

    if (clientID == null || clientID.isEmpty() || clientSecret == null || clientSecret.isEmpty()) {
      System.err.println(
          "GitHub authorization is not set. Please refer to the project README to"
              + " configure this. Using unauthorized GitHub API for now.");
      return GitHub.connectAnonymously();
    }

    authenticatedGitHub = new GitHubBuilder().withPassword(clientID, clientSecret).build();

    try {
      gitHub.checkApiUrlValidity();
      return authenticatedGitHub;
    } catch (IOException e) {
      System.err.println(
          "Invalid GitHub credentials. Please refer to the project README to"
              + " configure this. Using unauthorized GitHub API for now.");
     return GitHub.connectAnonymously();
```

Comment (josephdhurst) [reply src/main/java/com/google/opensesame/github/GitHubGetter.java:51]: Also, if possible test `buildGitHubInterface`. To make `buildGitHubInterface` easier to test you might want to make the `System.getenv` calls in getGitHub and then pass the results as arguments to `buildGitHubInterface`.

Comment (Richie78321) [reply src/main/java/com/google/opensesame/github/GitHubGetter.java]: Totally agree with both comments. Thanks! Will add some testing to this.

File: src/main/webapp/WEB-INF/appengine-web.xml [hunk 2]
```diff
@@ -7,4 +7,8 @@
     <!-- prevent unwanted caching when accessing via the web preview server -->
     <include path="/**" expiration="0s" />
   </static-files>
+  <env-variables>
+    <env-var name="GITHUB_CLIENT_ID" value="" />
+    <env-var name="GITHUB_CLIENT_SECRET" value="" />
+  </env-variables>
```
[the remark this answers is not in the document]
Comment (Richie78321) [reply src/main/webapp/WEB-INF/appengine-web.xml:13]: If they were filled in, our credentials would be public-facing (as all of this code is viewable by the internet), which would be very bad haha.

[the remark this answers is not in the document]
Comment (Richie78321) [reply src/main/webapp/WEB-INF/appengine-web.xml:13]: I'm hoping that the checks and alerts in the `GitHubGetter` will help us to not forget. It works just fine without them. Really the only time that we need to remember is when we're deploying to production!

File: src/main/java/com/google/opensesame/github/GitHubGetter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    GitHub authenticatedGitHub = new GitHubBuilder().withPassword(clientId, clientSecret).build();
+
+    try {
+      gitHub.checkApiUrlValidity();
+      return authenticatedGitHub;
+    } catch (IOException e) {
+      System.err.println(
+          "Invalid GitHub credentials. Please refer to the project README to"
+              + " configure this. Using unauthorized GitHub API for now.");
+      return GitHub.connectAnonymously();
```
[the remark this answers is not in the document]
Comment (Richie78321) [reply src/main/java/com/google/opensesame/github/GitHubGetter.java:51]: Thanks!

**Pull request merged**